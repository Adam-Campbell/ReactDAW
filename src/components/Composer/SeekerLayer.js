import React from 'react';
import PropTypes from 'prop-types';
import { Layer, Line } from 'react-konva';
import { UIColors } from '../../constants';

const SeekerLayer = props => (
    <Layer ref={props.seekerLayerRef}>
        <Line
            ref={props.seekerLineRef}
            points={[0, 0, 0, props.canvasHeight]}
            listening={false}
            stroke={UIColors.offWhite}
            strokeWidth={2}
        />
    </Layer>
);

SeekerLayer.propTypes = {
    seekerLayerRef: PropTypes.object.isRequired,
    seekerLineRef: PropTypes.object.isRequired,
    canvasHeight: PropTypes.number.isRequired
}

export default SeekerLayer;