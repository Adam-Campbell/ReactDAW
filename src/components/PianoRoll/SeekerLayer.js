import React from 'react';
import { Layer, Line } from 'react-konva';
import PropTypes from 'prop-types';
import { UIColors } from '../../constants';

const SeekerLayer = props => (
    <Layer
        x={48}
        ref={props.seekerLayerRef}
    >
        <Line
            ref={props.seekerLineRef}
            points={[0, 0, 0, props.canvasHeight]}
            listening={false}
            stroke={UIColors.offWhite}
            strokeWidth={0}
        />
    </Layer>
);

SeekerLayer.propTypes = {
    seekerLayerRef: PropTypes.object.isRequired,
    seekerLineRef: PropTypes.object.isRequired,
    canvasHeight: PropTypes.number.isRequired
};

export default SeekerLayer;