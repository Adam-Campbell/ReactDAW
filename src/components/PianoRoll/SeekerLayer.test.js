import React from 'react';
import { shallow } from 'enzyme';
import SeekerLayer from './SeekerLayer';

test('renders correctly', () => {
    const component = shallow(
        <SeekerLayer 
            seekerLayerRef={React.createRef()}
            seekerLineRef={React.createRef()}
            canvasHeight={1768}
        />
    );
    expect(component).toMatchSnapshot();
}) 