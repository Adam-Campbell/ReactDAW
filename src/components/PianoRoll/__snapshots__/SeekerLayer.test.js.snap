// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Layer
  x={48}
>
  <Line
    listening={false}
    points={
      Array [
        0,
        0,
        0,
        1768,
      ]
    }
    stroke="#e0e0e0"
    strokeWidth={0}
  />
</Layer>
`;
